src/main/java/wheresmymoney/RecurringExpense.java

```diff
@@ -0,0 +1,48 @@
+package wheresmymoney;
+
+import wheresmymoney.exception.WheresMyMoneyException;
+
+public class RecurringExpense extends Expense {
+    protected String frequency;
+    private String lastDateAdded;
+
+    public RecurringExpense(Float price, String description, String category, String frequency) 
+            throws WheresMyMoneyException {
+        super(price, description, category);
+        this.lastDateAdded = DateUtils.dateFormatToString(DateUtils.getCurrentDate());
+        this.frequency = frequency;
+    }
+
+    public RecurringExpense(Float price, String description, String category, 
+            String lastAddedDate, String frequency) throws WheresMyMoneyException{
+        super(price, description, category, lastAddedDate);
+        this.lastDateAdded = lastAddedDate;
+        this.frequency = frequency;
+    }
+
+    public String getlastAddedDate() {
+        return this.lastDateAdded;
+    }
+
+    public String getFrequency() {
+        return this.frequency;
+    }
+
+    public void setLastAddedDate(String date) throws WheresMyMoneyException {
+        if (date == null) {
+            throw new WheresMyMoneyException("Date should not be null");
```

Comment:
Can change to InvalidInputException in the future (for now this is fine, not too urgent)